lib/rubocop/cop/style/redundant_filter_chain.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Identifies usages of `any?`, `empty?` or `none?` predicate methods
+      # chained to `select`/`filter` and change them to use predicate method instead.
+      #
+      # @example
+      #   # bad
+      #   arr.select { |x| x > 1 }.any?
+      #
+      #   # good
+      #   arr.any? { |x| x > 1 }
+      #
+      #   # bad
+      #   arr.select { |x| x > 1 }.empty?
+      #   arr.select { |x| x > 1 }.none?
+      #
+      #   # good
+      #   arr.none? { |x| x > 1 }
+      #
+      #   # good
+      #   relation.select(:name).any?
+      #   arr.select { |x| x > 1 }.any?(&:odd?)
+      #
+      class RedundantFilterChain < Base
+        extend AutoCorrector
+
+        MSG = 'Use `%<prefer>s` instead of `%<first_method>s.%<second_method>s`.'
+
+        RAILS_METHODS = %i[many?].freeze
+        RESTRICT_ON_SEND = (%i[any? empty? none? one?] + RAILS_METHODS).freeze
+
+        # @!method select_predicate?(node)
+        def_node_matcher :select_predicate?, <<~PATTERN
+          (send
+            {
+              (block $(send _ {:select :filter}) ...)
+              $(send _ {:select :filter} block_pass_type?)
+            }
+            ${:#{RESTRICT_ON_SEND.join(' :')}})
+        PATTERN
+
+        REPLACEMENT_METHODS = {
```

Comment:
Might be better to make this configurable instead of hardcoded. This would also remove the need to hardcode the support for `many?`.

Reply:
(whether it's useful to do this depends mostly on how common is for people to encounter other similar methods in the wild)

Reply:
We need a consensus https://github.com/rubocop/rubocop-performance/pull/342#issuecomment-1531080588 😅

Reply:
I'm fine with @koic's suggestion.

Reply:
It seem the PR currently implements it, right?

Reply:
Yes. Expanded docs with an additional example.